Gems/Pointcloud/Code/Source/Tools/Components/PointcloudEditorComponent.cpp

```diff
@@ -0,0 +1,173 @@
+#include <AzToolsFramework/UI/UICore/WidgetHelpers.h>
+#include <Render/PointcloudFeatureProcessor.h>
+namespace Pointcloud
+{
+
+    void PointcloudEditorComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("TransformService"));
+    }
+
+    void PointcloudEditorComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<PointcloudEditorComponent, AzToolsFramework::Components::EditorComponentBase>()
+                ->Version(2)
+                ->Field("Point Size", &PointcloudEditorComponent::m_pointSize)
+                ->Field("PointcloudAsset", &PointcloudEditorComponent::m_pointcloudAsset)
+                ->Field("Visible", &PointcloudEditorComponent::m_visible)
+                ->Field("NumPoints", &PointcloudEditorComponent::m_numPoints);
+            AZ::EditContext* editContext = serializeContext->GetEditContext();
+            if (editContext)
+            {
+                editContext->Class<PointcloudEditorComponent>("PointcloudEditorComponent", "PointcloudEditorComponent")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "PointcloudEditorComponent")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "RobotecTools")
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &PointcloudEditorComponent::m_pointSize,
+                        "Point Size",
+                        "Size of the points in the pointcloud")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &PointcloudEditorComponent::OnSetPointSize)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &PointcloudEditorComponent::m_visible, "Visible", "Visibility of the pointcloud")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &PointcloudEditorComponent::OnVisibility)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &PointcloudEditorComponent::m_pointcloudAsset,
+                        "Pointcloud Asset",
+                        "Asset containing the pointcloud data")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &PointcloudEditorComponent::m_numPoints,
+                        "NumPoints",
+                        "Number of points in the pointcloud")
+                    ->Attribute(AZ::Edit::Attributes::ReadOnly, true)
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &PointcloudEditorComponent::LoadCloud);
```

Comment:
I don't get the ChangeNotify purpose here. When it is supposed to trigger?

Reply:
It has no sense and purpouse